### PR TITLE
fix(config) Support join relationships in configuration

### PR DESCRIPTION
**Pull request by evanh**
Add support to the schemas to allow entities to specify join relationships.
Also add a test for the join relationship parser.

Since CDC is a deprecated feature, there was a discussion about whether we should support joins in config, since CDC is the only user of them. But I don't think we want to eliminate support for joins entirely, so I decided to add support for the relationships. Perhaps in the future someone will want joins and we will be ready.

### CODE REVIEW

File: snuba/datasets/configuration/entity_builder.py [hunk 1]
```diff
@@ -136,8 +170,9 @@ def build_entity_from_config(file_path: str) -> PluggableEntity:
         required_time_column=config["required_time_column"],
         validators=_build_entity_validators(config["validators"]),
         translation_mappers=_build_entity_translation_mappers(
-            config["translation_mappers"]
+            config.get("translation_mappers", {})
```
Comment (evanh) [inline snuba/datasets/configuration/entity_builder.py:173]: `translation_mappers` is not a required field in the schema, and in fact some entities (CDC) don't specify them. So I changed this to handle entities that don't specify any mappers.

File: snuba/datasets/configuration/json_schema.py [hunk 2]
```diff
@@ -442,6 +442,50 @@ def registered_class_array_schema(
     },
 }
 
+ENTITY_JOIN_RELATIONSHIPS = {
+    "type": "object",
+    "patternProperties": {
```
Comment (evanh) [inline snuba/datasets/configuration/json_schema.py:447]: This allows arbitrary keys with specific values.

File: snuba/datasets/configuration/json_schema.py [hunk 3]
```diff
@@ -442,6 +442,50 @@ def registered_class_array_schema(
...
+            "properties": {
+                "rhs_entity": {
+                    "type": "string",
+                    "description": "The entity key of the rhs entity to join with",
+                },
+                "columns": {
+                    "type": "array",
+                    "items": {
+                        "type": "array",
+                        "prefixItems": [
```
Comment (evanh) [inline snuba/datasets/configuration/json_schema.py:460]: `prefixItems` lets you do tuple style validation.

File: tests/datasets/configuration/entity_join_relationships.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+version: v1
```
Comment (evanh) [inline tests/datasets/configuration/entity_join_relationships.yaml:1]: This is actually very close to what the groupassignee schema would actually be.

**Review by enochtangg — APPROVED** [review]

**Pull request merged**